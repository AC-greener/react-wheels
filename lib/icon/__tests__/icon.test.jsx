import renderer from 'react-test-renderer'
import React from 'react'
import Icon from '../icon'
import { mount } from 'enzyme'
describe('Icon', () => {

  it('should render success', () => {
    const json = renderer.create(<Icon name='wechat'/>).toJSON()
    expect(json).toMatchSnapshot()
  })

  it('should recive prop: name', () => {
    const component = mount(<Icon name="alipay"/>)
    expect(component.props().name).toEqual("alipay")
    component.setProps({name: "wechat"})
    expect(component.props().name).toEqual("wechat")
  })

  it('should render with primary color', () => {
    const component = mount(<Icon color="primary"/>)
    expect(component.find('svg').hasClass("color-primary")).toEqual(true)
  })

  it('should clicked', () => {
    let n = 1
    const fn = () => {n += 1}
    const component = mount(<Icon name="alipay" onClick={fn}/>)
    component.find('svg').simulate('click')
    expect(n).toEqual(2)
  })

  it('should fake clicked', () => {
    const fn = jest.fn()
    const component = mount(<Icon name="alipay" onClick={fn}/>)
    component.find('svg').simulate('click')
    expect(fn).toBeCalled()
  })
})