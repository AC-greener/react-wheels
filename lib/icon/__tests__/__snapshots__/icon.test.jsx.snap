// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon should render success 1`] = `
<svg
  className="xui-icon"
>
  <use
    xlinkHref="#wechat"
  />
</svg>
`;
